lib/doorkeeper/oauth/refresh_token_request.rb

```diff
@@ -35,8 +35,10 @@ def before_successful_response
         refresh_token.transaction do
           refresh_token.lock!
           raise Errors::InvalidTokenReuse if refresh_token.revoked?
+          if refresh_token.revoked_at.nil?
+            refresh_token.revoke_in(server.refresh_token_revoked_in)
+          end
 
-          refresh_token.revoke
```

Comment:
Actually we need the new condition here to not revoke refresh_token if the new config is used.